opencensus_shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetryContextManager.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.opencensusshim;
+
+import io.opencensus.trace.ContextHandle;
+import io.opencensus.trace.ContextManager;
+import io.opencensus.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.trace.TracingContextUtils;
+
+public class OpenTelemetryContextManager implements ContextManager {
```

Comment:
Have you considered the opposite approach, having OTel context use the OpenCensus context? Scope is more flexible than returning previousCtx, so the bridge can be a bit cleaner than the other way around. I suspect this code could be adapted to opencensus context trivially.

https://github.com/open-telemetry/opentelemetry-java/blob/master/context/src/otelAsGrpcTest/java/io/opentelemetry/context/GrpcContextStorageProvider.java

Reply:
Yep, decided on the current approach as we're trying to migrate everyone away from OpenCensus. Having OTel context use the OpenCensus context would mean that applications and libraries would have to configure OpenCensus rather than OpenTelemetry, which is something we want to avoid. The goal is to remove this once everyone moves to OpenTelemetry.

Reply:
My understanding is that with @anuraaga 's suggestion we would still use the OpenTelemetry pipeline at a high level, though it would be storing context in OpenCensus.

I still do agree this would be more confusing than migrating everything to OpenTelemetry like it's done in this PR.

Side effects of this shim only affect code instrumented with OpenCensus, not touching parts instrumented with OpenTelemetry at all.

Reply:
I guess this shim by nature assumes OC is there - so I think having the OTel context implementation also assume that should be ok? Once OC is removed from the classpath it'd just be normal OTel with no change in the app.

This approach seems fine too though, so just wanted to point it out, the main advantage would just be that the `.scope` stuff below could be a bit simpler.

Reply:
Crosstalk :) But yeah I think either approach is fine if they can satisfy bidirectional interop like in 

https://github.com/open-telemetry/opentelemetry-java/blob/master/context/src/otelAsGrpcTest/java/io/opentelemetry/context/OtelAsGrpcTest.java

I know the OTelAsOC approach would work, but not sure about the reverse until we try :)

Reply:
> I guess this shim by nature assumes OC is there

Actually no - this shim will work fine for libraries instrumented with OpenCensus API (e.g. Google Cloud Spanner client) even if OpenCensus is not configured in the application.

i.e. the assumption is that OpenCensus "Implementation" pipeline has been replaced with OpenTelemetry but there are certain cases where OpenCensus "API" is still used, so shim helps with implicitly exporting those into OpenTelemetry as well.

Reply:
Ah got it - thanks for the explanation